From: Peter Samuelson <peter@p12n.org>
Date: Thu, 05 Jan 2006 10:30:35 -0600
Subject: Eliminate several unneeded inter-library linkages

More link pruning, which eliminates one package dependency (subversion
-> libneon).  I'll probably submit at least the build.conf portion of
this upstream.

[ps, 2006-09-09: Remove a couple more instances of neon (nobody but
 libsvn_ra_dav should link neon directly).  Also hack the apriconv
 library declaration not to pull in aprutil.  (It might be correct on
 Windows, I don't know, but we don't need apriconv _at all_, so we
 don't want it to pull in aprutil for us.)]

[ps, 2010-11-08: refresh]

Index: subversion-1.7.0/build.conf
===================================================================
--- subversion-1.7.0.orig/build.conf	2011-10-20 21:55:44.000000000 +0200
+++ subversion-1.7.0/build.conf	2011-10-20 21:55:48.000000000 +0200
@@ -109,7 +109,7 @@
 type = exe
 path = subversion/svn
 libs = libsvn_client libsvn_wc libsvn_ra libsvn_delta libsvn_diff libsvn_subr
-       apriconv apr neon
+       apriconv apr
 manpages = subversion/svn/svn.1
 install = bin
 
@@ -154,7 +154,7 @@
 description = Subversion repository replicator
 type = exe
 path = subversion/svnsync
-libs = libsvn_ra libsvn_delta libsvn_subr apr neon
+libs = libsvn_ra libsvn_delta libsvn_subr apr
 install = bin
 manpages = subversion/svnsync/svnsync.1
 
@@ -597,7 +597,7 @@
 type = lib
 path = subversion/bindings/javahl/native
 libs = libsvn_repos libsvn_client libsvn_wc libsvn_ra libsvn_delta libsvn_diff 
-       libsvn_subr libsvn_fs aprutil apriconv apr neon
+       libsvn_subr libsvn_fs apriconv apr
 sources = *.cpp *.c
 add-deps = $(javahl_javah_DEPS) $(javahl_java_DEPS) $(javahl_callback_javah_DEPS) $(javahl_types_javah_DEPS)
 install = javahl-lib
@@ -1023,7 +1023,7 @@
 
 [apriconv]
 type = lib
-external-lib = $(SVN_APRUTIL_LIBS)
+external-lib = $(SVN_APR_LIBS)
 
 [bdb]
 type = lib
Index: subversion-1.7.0/subversion/bindings/swig/perl/native/Makefile.PL.in
===================================================================
--- subversion-1.7.0.orig/subversion/bindings/swig/perl/native/Makefile.PL.in	2011-10-20 20:46:13.000000000 +0200
+++ subversion-1.7.0/subversion/bindings/swig/perl/native/Makefile.PL.in	2011-10-20 21:55:48.000000000 +0200
@@ -42,7 +42,12 @@
                    @modules, qw/diff subr ra_local ra_svn ra_neon
                                 ra_serf fs_base fs_util fs_fs/);
 
-my @ldmodules = map {"-lsvn_$_-1"} (@modules, qw/diff subr/);
+sub libs_line {
+   return join(' ', $apr_ldflags,
+               (map {$_ = abs_path($_); "-L$_"} @ldpaths),
+               (map {"-lsvn_$_-1"} @_),
+               '-lsvn_swig_perl-1', `$swig -perl -ldflags`);
+}
 
 my $apr_shlib_path_var = '@SVN_APR_SHLIB_PATH_VAR@';
 my $cppflags = '@CPPFLAGS@';
@@ -67,10 +72,6 @@
                  " -I$svnlib_builddir",
                  " -I$swig_srcdir -g"),
     OBJECT => q/$(O_FILES)/,
-    LIBS => [join(' ', $apr_ldflags,
-                  (map {"-L$_"} @ldpaths),
-                  @ldmodules, '-lsvn_swig_perl-1',
-                  `$swig -perl -ldflags`)],
     test => { TESTS => "$swig_srcdir/perl/native/t/*.t" }
 );
 
@@ -80,6 +81,7 @@
     $_;
 }
 
+$config{LIBS} = [libs_line('diff', 'subr')];
 WriteMakefile(%config, NAME => 'SVN::_Core', C => ['core.c'],
         PM => {map { ("$swig_srcdir/perl/native/$_.pm" =>
                       "\$(INST_LIBDIR)/$_.pm") }
@@ -94,6 +96,7 @@
        );
 
 for (@modules) {
+    $config{LIBS} = [libs_line($_, 'subr')];
     WriteMakefile(%config,
       MAKEFILE=> "Makefile.$_",
       NAME    => "SVN::_".perlish($_),
